tests/test_hooks/test_cmd.py

```diff
@@ -11,27 +10,34 @@
 
 class TestCmd(object):
     def setup_method(self, test_method):
-        self.stack = Mock(Stack)
-        self.stack.name = "my/stack.yaml"
+        self.stack = Stack(
+            "stack1",
+            "project1",
+            "region1",
+            template_handler_config={"template": "path.yaml"},
+        )
+
+        # Otherwise the test works only when the environment variables already
+        # set a valid AWS session.
+        self.stack.connection_manager.create_session_environment_variables = Mock(
+            return_value={}
+        )
+
         self.cmd = Cmd(stack=self.stack)
 
     def test_run_with_non_str_argument(self):
         self.cmd.argument = None
         with pytest.raises(InvalidHookArgumentTypeError):
             self.cmd.run()
 
-    @patch("sceptre.hooks.cmd.subprocess.check_call")
-    def test_run_with_str_argument(self, mock_call):
+    def test_run_with_str_argument(self, capfd):
```

Comment:
remember to test running with a dict arg as well.

Reply:
Yes, I'm looking at how to do that. I've tried to make my tests depend only on the behavior and not the implementation. Just now the test doesn't know that subprocess is being used.

But I don't know how to to test that the `executable` option really causes a different shell to be launched.

I think I might have to use a mock after all and make the tests depend on the `subprocess` module. That module is the obvious way to implement this hook, so maybe it's not so bad.